locationchanger

```diff
@@ -3,27 +3,33 @@
 # automatically change configuration of Mac OS X based on location
 exec &>/usr/local/var/log/locationchanger.log
 # redirect all IO to /dev/null (comment this out if you want to debug)
-# exec 1>/dev/null 2>/dev/null
+#exec 1>/dev/null 2>/dev/null
 
 # get a little breather before we get data for things to settle down
 sleep 2
 
+# Escape problematic caracters before passing strings to egrep or grep
+# see discussion on:
+# https://stackoverflow.com/questions/11856054/bash-easy-way-to-pass-a-raw-string-to-grep
+egrep_quote() { sed 's/[]\.|$(){}?+*^]/\\&/g' <<< "$*" }
+grep_quote() { sed 's/^$[]*.\/\\&/g' <<< "$*" }
+
 # get various system information
 SSID=`/System/Library/PrivateFrameworks/Apple80211.framework/Versions/A/Resources/airport -I\
- | grep ' SSID:' | cut -d ':' -f 2 | tr -d ' '`
+ | grep ' SSID:' | sed s/.*SSID:\ // | tr -d ' '`
 
 # basic pattern matching to get all location names separated by newlines
-LOCATION_NAMES=`scselect | tail -n +2 | cut -d "(" -f 2 | cut -d ")" -f 1`
```

Comment:
This would fail if location name contained "(" or ")" character

Reply:
Thanks, for the fix! 👍